ptype/data.py

```diff
@@ -0,0 +1,69 @@
+import os
+import random
+import pandas as pd
+from tqdm import tqdm
+from sklearn.preprocessing import StandardScaler, MinMaxScaler
+
+
+def load_ptype_data(data_path, source, train_val_test_proportion=[70,20,10], seed=1):
+    """
+    Load Precip Type data
+    Args:
+        data_path (str): Path to data
+        source (str): Precip observation source. Supports 'ASOS' or 'mPING'.
+        train_val_test_proportion (list of int): proportion of data to use in each train, validation, and test split.
+        seed (int): random seed used in random shuffle.
+    Returns:
+    Dictionary of Pandas dataframes of training / validation / test data
+    """
+    
+    dates = [x[-16:-8] for x in os.listdir(os.path.join(data_path, source))]
+    random.Random(seed).shuffle(dates)
+    
```

Comment:
I think this is a scenario where we don't want completely random shuffles between the different splits due to temporal contamination. An easy fix with the current code would be to sort the dates and remove the shuffle.

`dates = sorted([x[-16:-8] for x in os.listdir(os.path.join(data_path, source))])`

However, you might also consider using a date sorter so you can specific specific ranges for each split. That would probably require running the function for each split using a `start_date` and `end_date` parameter and then converting them to strings with something like 

`pd.date_range(start_date, end_date, freq='h').strftime("%Y%m%d%")`

Reply:
Thanks Charlie - I thought that temporal contamination was important to worry about within the same day but not among days within the same year/month/season? In any case, I've updated the code to reflect start and end dates.